metadata-ingestion/src/datahub/ingestion/source/tableau_common.py

```diff
@@ -762,8 +762,19 @@ def make_upstream_class(
 
 
 def make_fine_grained_lineage_class(
-    parsed_result: Optional[SqlParsingResult], dataset_urn: str
+    parsed_result: Optional[SqlParsingResult],
+    dataset_urn: str,
+    out_columns: List[Dict[Any, Any]],
 ) -> List[FineGrainedLineage]:
+    # 1) fine grane linage links seems like case sensitive
+    # 2) parsed out columns are always lower cased
+    # 3) corresponding Custom SQL output columns can be in any case lower/upper/mix
+    #
+    # we need a map between 2 and 3 that will be used during building column level linage links (see below)
+    out_columns_map = {
+        col.get(c.NAME, "").casefold(): col.get(c.NAME, "") for col in out_columns
```

Comment:
I haven't really come across casefold before - is there a strong reason to use it instead of .lower()?

Reply:
@hsheth2 yep, i assumed that columns may contain some language-specific characters, casefold() handles it properly, but i'm switching it back for consistency, if everywhere .lower() is used then better to use .lower() here as well.